### ISSUE TITLE
[Bug]: Using `bin/console cache:warmup` gives warning of FrozenBag

### ISSUE THREAD

**Issue by sauls** [Unleash/unleash-client-symfony#49]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

![image](https://github.com/Unleash/unleash-client-symfony/assets/7901434/fdb84d3c-5b28-487b-a1c7-deb9b4835a97)

**Edit**: the warning message in the image reads: 
> WARNING [app] Failed to generate ConfigBuilder for extension Unleash\Client\Bundle\DependencyInjection\UnleashSymfonyClientExtension: Impossible to call set() on a frozen ParameterBag. ["exception" => Symfony\Component\DependencyInjection\Exception\LogicException { …},"extensionClass" => "Unleash\Client\Bundle\DependencyInjection\UnleashSymfonyClientExtension"]

### To reproduce

Running `bin/console cache:warmup` gives this warning

### Version

v0.10.282

### Expected behavior

This warning should not be thrown.

**Comment by thomasheartman** [comment]
Hi, @sauls. Thanks for reporting this issue. I've taken the liberty of copying the text from your image into the issue description so it's easier to see what the warning is. 

From my understanding, this is just a warning and not something that stops the SDK from working in any way, is that correct? Or are you seeing any adverse effects? 

Anyway, I'd like to hear what @RikudouSage has to say about this as well. It might be an easy fix or it might be a third-party dependency that we can't touch.

**Comment by RikudouSage** [comment]
@thomasheartman I'm not entirely sure this is something on our side. 

@sauls Any chance you could create a repo where I can reproduce this? This is a very generic error and sadly the true cause of the error is hidden by Symfony. If you search for the error in Symfony source code, you'll find it in `ConfigBuilderCacheWarmer` and you'll see that it's in an exception catch block. I'd need to know what the actual error is from the exception.

**Comment by sauls** [comment]
> Hi, @sauls. Thanks for reporting this issue. I've taken the liberty of copying the text from your image into the issue description so it's easier to see what the warning is.
> 
> From my understanding, this is just a warning and not something that stops the SDK from working in any way, is that correct? Or are you seeing any adverse effects?
> 
> Anyway, I'd like to hear what @RikudouSage has to say about this as well. It might be an easy fix or it might be a third-party dependency that we can't touch.

Well no, everything works well, but in CI/CD pipeline seeing warnings indicates potentially something is not right :) so I like to have lets call it "clean" runs where no warning nor any level errors are shown. And if they are shown then I need to react. But that just personal thing..

> @thomasheartman I'm not entirely sure this is something on our side.
> 
> @sauls Any chance you could create a repo where I can reproduce this? This is a very generic error and sadly the true cause of the error is hidden by Symfony. If you search for the error in Symfony source code, you'll find it in `ConfigBuilderCacheWarmer` and you'll see that it's in an exception catch block. I'd need to know what the actual error is from the exception.

@thomasheartman @RikudouSage I ported out my project unleash configuration to a fresh Symfony 6.4 install and I can confirm that there is no issue with it. I guess I need to investigate deeper in my project why this happens. 

Thank you for your time.

**Comment by RikudouSage** [comment]
Feel free to reopen if it turns out to be an Unleash issue.

**Comment by thomasheartman** [comment]
> I like to have lets call it "clean" runs where no warning nor any level errors are shown. And if they are shown then I need to react. But that just personal thing.

@sauls Oh, of course. I think we could all do with fewer warnings in our lives 😅  It was more a matter of determining the potential urgency of the issue if it were the SDK: if something were broken, then that's obviously more pressing than a rogue warning.

Anyway, thanks for investigating deeper and yeah, do let us know if it turns out to be on Unleash ☺️

**Comment by Okhoshi** [comment]
Hey 👋 

I can reproduce the problem, with a brand new project.

```
symfony new --dir unleash_cache_clear_repro
cd unleash_cache_clear_repro
composer require unleash/symfony-client-bundle
bin/console cache:clear -v
```

I investigated a bit the cause of the error, and it's because in the extension class, `getConfiguration` method loads the `services.yaml` file, and since this file contains a parameter definition, it fails to insert it again as the container was already compiled and frozen in certain situations.

A possible fix that I tried and worked successfully is to move the `parameters` section to a dedicated file that would not be loaded in the `getConfiguration` method, only the `load` method (similar to `autowiring.yaml`).

However, it would be better to not load the `services.yaml` file at all IMHO, because it makes it a pain to override any definition set by the Extension, because `getConfiguration` is called way later than `load` in the chain. Especially since it's only used to fill-in an informative message, that could have been static.

I can provide whatever fix you prefer in a PR, just let me know which one I should submit :)

Thank you for your time and all the efforts on Unleash ! :)
